aws/product/awsec2product.go

```diff
@@ -0,0 +1,179 @@
+package product
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/trackit/jsonlog"
+
+	"github.com/trackit/trackit2/config"
+	"github.com/trackit/trackit2/db"
+	"github.com/trackit/trackit2/models"
+)
+
+type (
+	PricingSku string
+
+	PricingAttribute struct {
+		InstanceType       string
+		CurrentGeneration  string
+		Vcpu               string
+		Memory             string
+		Storage            string
+		NetworkPerformance string
+		Tenancy            string
+		OperatingSystem    string
+		Ecu                string
+		Location           string
+		LocationType       string
+	}
+
+	PricingProduct struct {
+		Sku           PricingSku
+		ProductFamily string
+		Attributes    PricingAttribute
+	}
+
+	EC2Pricing struct {
+		Products map[PricingSku]PricingProduct
+	}
+)
+
+func storeRegionID(region string, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger, sqlTx models.XODB) error {
+	dbAwsRegion, err := models.AwsRegionByPretty(sqlTx, region)
+	if err != nil && err.Error() == "sql: no rows in result set" {
+		var newDbAwsRegion models.AwsRegion
+		newDbAwsRegion.Pretty = region
```

Comment:
Why the extra “pretty” field if it’s just the same value ? This seems useless and the sole justification for creating a new table for regions. Adding a join is adding complexity, and it seems we do not need it at all.